Subject: config.sub: accept substrate as an OS name

Wire the substrate* OS family into the top-level config.sub so triples like i386-unknown-substrate and x86_64-unknown-substrate stop bouncing off the unrecognised-OS check at configure time.

---
diff --git a/config.sub b/config.sub
--- a/config.sub	2026-05-11 21:33:53.996778439 -0500
+++ b/config.sub	2026-05-11 21:34:25.498259917 -0500
@@ -2115,6 +2115,7 @@
 	| solidbsd* \
 	| sortix* \
 	| storm-chaos* \
+	| substrate* \
 	| sunos \
 	| sunos[34]* \
 	| superux* \
